packages/babel-traverse/src/path/index.ts

```diff
@@ -52,35 +52,37 @@ export type NodeOrNodeList<T extends t.Node> = T | NodeList<T>;
 export type NodeList<T extends t.Node> = T[] | [T, ...T[]];
 
 const NodePath_Final = class NodePath {
-  constructor(hub: HubInterface, parent: t.Node | null) {
+  constructor(hub: HubInterface | undefined, parent: t.Node) {
     this.parent = parent;
-    this.hub = hub;
+    this.hub = hub!;
     this.data = null;
 
+    // @ts-expect-error Set it in setContext
     this.context = null;
+    // @ts-expect-error Set it in setContext
     this.scope = null;
   }
 
   declare parent: t.Node;
   declare hub: HubInterface;
-  declare data: Record<string | symbol, unknown>;
+  declare data: Record<string | symbol, unknown> | null;
   // TraversalContext is configured by setContext
   declare context: TraversalContext;
   declare scope: Scope;
 
   contexts: Array<TraversalContext> = [];
   state: any = null;
-  opts: ExplodedTraverseOptions | null = null;
+  declare opts: ExplodedTraverseOptions;
 
-  @bit.storage _traverseFlags: number;
+  @bit.storage _traverseFlags: number = 0;
   @bit(REMOVED) accessor removed = false;
   @bit(SHOULD_STOP) accessor shouldStop = false;
   @bit(SHOULD_SKIP) accessor shouldSkip = false;
 
   skipKeys: Record<string, boolean> | null = null;
-  parentPath: NodePath_Final = null;
+  parentPath: NodePath_Final | null = null;
   container: t.Node | Array<t.Node> | null = null;
-  listKey: string | null = null;
+  listKey: string | null | undefined = null;
   key: string | number | null = null;
   node: t.Node | null = null;
   type: t.Node["type"] | null = null;
```

Comment:
Perhaps we could define more properties as non-null, such as `container`, like we did with `parentPath`.